src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,104 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+    private static final int DEFAULT_SIZE = 10;
+    private T[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = (T[]) (new Object[DEFAULT_SIZE]);
+    }
 
+    public void add(T value) {
+        add(value, size);
```

Comment:
it's good that you try to reuse code. however, adding an element to the end of the list does not require copying the array to free space for a new element. So, your code is working but less performant

Reply:
You're right. Code edited